### PR TITLE
Add partitionBy columns option when writing delta table

### PR DESCRIPTION
**Pull request by ekstromchristian**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '3.8'
+version: '3.3'
```
Comment (bulv1ne) [inline docker-compose.yaml]: Why the change?

File: getl/blocks/write/entrypoint.py [hunk 2]
```diff
@@ -147,10 +150,14 @@ def batch_delta(conf: BlockConfig) -> DataFrame:
     """
     path = conf.get("Path")
     mode = conf.get("Mode")
+    columns = conf.get("PartitionBy.Columns", None)
     dataframe = conf.history.get(conf.input)
     batch = BatchDelta(dataframe, conf.spark)
     htable = None
 
+    if columns:
+        columns = columns.replace(" ", "").split(",")
```
Comment (bulv1ne) [inline getl/blocks/write/entrypoint.py]: Why not require columns to already be a list of strings? Like `write::batch_postgres_upsert` Columns?

File: getl/blocks/write/entrypoint.py [hunk 3]
```diff
@@ -129,6 +130,8 @@ def batch_delta(conf: BlockConfig) -> DataFrame:
         Properties:
             Path: s3://path/to/files
             Mode: upsert
+            PartitionBy:
+                Columns: year, month, day
```
Comment (bulv1ne) [inline getl/blocks/write/entrypoint.py]:
```suggestion
                Columns: [year, month, day]
```

Comment (bulv1ne) [reply getl/blocks/write/entrypoint.py]: This now is an array in yaml

File: tests/getl/blocks/write/test_write_entrypoint.py [hunk 4]
```diff
@@ -37,6 +40,34 @@ def test_batch_delta_overwrite(m_hive_table, helpers, spark_session, tmp_dir):
...
+@patch.object(BatchDelta, "write")
+def test_batch_delta_partitionby(
+    m_write, m_hive_table, helpers, spark_session, tmp_dir
+):
+    """Batch write delta files with partitionBy."""
+    # Arrange
+    props = {
+        "Path": tmp_dir,
+        "Mode": "overwrite",
+        "PartitionBy": {"Columns": "year, month"},
```
Comment (bulv1ne) [inline tests/getl/blocks/write/test_write_entrypoint.py]:
```suggestion
        "PartitionBy": {"Columns": ["year", "month"]},
```

File: docker-compose.yaml (hunk 1 above)
Comment (ekstromchristian) [reply docker-compose.yaml]: ERROR: Version in "./docker-compose.yaml" is unsupported. You might be seeing this error because you're using the wrong Compose file version. Either specify a supported version (e.g "2.2" or "3.3") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.
For more on the Compose file format versions, see https://docs.docker.com/compose/compose-file/

Comment (ekstromchristian) [reply docker-compose.yaml]: I can't start the db with 3.8.
docker-compose up -d db command fails

Comment (bulv1ne) [reply docker-compose.yaml]: @ekstromchristian do you have the latest docker-compose version?

https://docs.docker.com/compose/compose-file/

```
$ docker-compose --version
docker-compose version 1.27.4, build unknown
```

**Review by bulv1ne — APPROVED** [review]
With the exception of docker-compose version change, but that’s a very minor thing so I don’t care :)

**Review by bulv1ne — CHANGES_REQUESTED** [review]
You removed my CHANGELOG.md update 😢 
```
git cherry-pick 12046e8fc8be0b29a4dfb47404f0d53df7aa3c57
```
git cherry-pick to the resque 😃

**Review by bulv1ne — APPROVED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (ekstromchristian) [reply docker-compose.yaml]: docker-compose version 1.24.1, build 4667896b

Comment (bulv1ne) [reply docker-compose.yaml]: So either upgrade your local docker-compose or change to 3.3, we’re not really using any important features from the latest versions.

Comment (ekstromchristian) [reply docker-compose.yaml]: I did the upgrade and added this to contribute.md
